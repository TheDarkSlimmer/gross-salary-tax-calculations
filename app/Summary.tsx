import { Text, SafeAreaView } from "react-native";

function Summary() {
  return (
    <SafeAreaView>
      <Text>Summary</Text>
    </SafeAreaView>
  );
}

export default Summary;
