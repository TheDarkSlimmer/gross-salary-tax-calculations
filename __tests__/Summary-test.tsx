import React from "react";
import renderer from "react-test-renderer";
import Summary from "../app/Summary";
import AsyncStorage from "@react-native-async-storage/async-storage";

test("renders correctly", () => {
  const tree = renderer
    .create(<Summary />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
