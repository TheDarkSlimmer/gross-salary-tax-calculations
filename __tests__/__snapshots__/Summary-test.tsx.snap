// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<RCTSafeAreaView>
  <RCTScrollView>
    <View>
      <View
        style={
          {
            "display": "flex",
            "gap": 13,
            "justifyContent": "center",
            "marginTop": 30,
            "padding": 30,
            "textAlign": "center",
            "width": "100%",
          }
        }
      >
        <Text
          style={
            {
              "fontFamily": "Inter-Medium",
              "fontSize": 20,
              "fontWeight": "600",
              "margin": "auto",
            }
          }
        >
          Your extimated gross salary is
        </Text>
        <Text
          style={
            {
              "fontFamily": "Inter-Medium",
              "fontSize": 30,
              "fontWeight": "600",
              "margin": "auto",
            }
          }
        >
          GHS0.00
        </Text>
      </View>
      <View
        style={
          {
            "borderColor": "#0077B6",
            "borderRadius": 5,
            "borderWidth": 2,
            "margin": "auto",
            "marginBottom": 20,
            "padding": 25,
            "width": "80%",
          }
        }
      >
        <View
          style={
            {
              "display": "flex",
              "flexDirection": "row",
              "fontWeight": "600",
              "gap": 20,
              "justifyContent": "space-between",
              "marginBottom": 20,
            }
          }
        >
          <Text
            style={
              {
                "fontFamily": "Inter",
                "fontSize": 15,
                "fontWeight": 600,
                "lineHeight": 20,
                "width": "50%",
              }
            }
          >
            Basic Salary
          </Text>
          <Text
            style={
              {
                "fontFamily": "Inter",
                "fontSize": 15,
                "fontWeight": 600,
              }
            }
          >
            GHS0.00
          </Text>
        </View>
        <View
          style={
            {
              "display": "flex",
              "flexDirection": "row",
              "fontWeight": "600",
              "gap": 20,
              "justifyContent": "space-between",
              "marginBottom": 20,
            }
          }
        >
          <Text
            style={
              {
                "fontFamily": "Inter",
                "fontSize": 15,
                "fontWeight": 600,
                "lineHeight": 20,
                "width": "50%",
              }
            }
          >
            Employee Pension Contribution Amount
          </Text>
          <Text
            style={
              {
                "fontFamily": "Inter",
                "fontSize": 15,
                "fontWeight": 600,
              }
            }
          >
            GHS0.00
          </Text>
        </View>
        <View
          style={
            {
              "display": "flex",
              "flexDirection": "row",
              "fontWeight": "600",
              "gap": 20,
              "justifyContent": "space-between",
              "marginBottom": 20,
            }
          }
        >
          <Text
            style={
              {
                "fontFamily": "Inter",
                "fontSize": 15,
                "fontWeight": 600,
                "lineHeight": 20,
                "width": "50%",
              }
            }
          >
            Employer Pension Amount
          </Text>
          <Text
            style={
              {
                "fontFamily": "Inter",
                "fontSize": 15,
                "fontWeight": 600,
              }
            }
          >
            GHS0.00
          </Text>
        </View>
        <View
          style={
            {
              "display": "flex",
              "flexDirection": "row",
              "fontWeight": "600",
              "gap": 20,
              "justifyContent": "space-between",
              "marginBottom": 20,
            }
          }
        >
          <Text
            style={
              {
                "fontFamily": "Inter",
                "fontSize": 15,
                "fontWeight": 600,
                "lineHeight": 20,
                "width": "50%",
              }
            }
          >
            Total PAYE Tax
          </Text>
          <Text
            style={
              {
                "fontFamily": "Inter",
                "fontSize": 15,
                "fontWeight": 600,
              }
            }
          >
            GHS0.00
          </Text>
        </View>
      </View>
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "alignItems": "center",
            "backgroundColor": "#0077B6",
            "borderRadius": 5,
            "borderWidth": 0,
            "display": "flex",
            "height": 38,
            "justifyContent": "center",
            "marginLeft": "auto",
            "marginRight": "auto",
            "opacity": 1,
            "width": 140,
          }
        }
      >
        <Text
          style={
            {
              "color": "white",
              "fontFamily": "Inter",
              "fontWeight": "600",
            }
          }
        >
          Calculate again
        </Text>
      </View>
    </View>
  </RCTScrollView>
</RCTSafeAreaView>
`;

exports[`retrieves userName from AsyncStorage and renders correctly 1`] = `
<RCTSafeAreaView>
  <RCTScrollView>
    <View>
      <View
        style={
          {
            "display": "flex",
            "gap": 13,
            "justifyContent": "center",
            "marginTop": 30,
            "padding": 30,
            "textAlign": "center",
            "width": "100%",
          }
        }
      >
        <Text
          style={
            {
              "fontFamily": "Inter-Medium",
              "fontSize": 20,
              "fontWeight": "600",
              "margin": "auto",
            }
          }
        >
          Your extimated gross salary is
        </Text>
        <Text
          style={
            {
              "fontFamily": "Inter-Medium",
              "fontSize": 30,
              "fontWeight": "600",
              "margin": "auto",
            }
          }
        >
          GHS0.00
        </Text>
      </View>
      <View
        style={
          {
            "borderColor": "#0077B6",
            "borderRadius": 5,
            "borderWidth": 2,
            "margin": "auto",
            "marginBottom": 20,
            "padding": 25,
            "width": "80%",
          }
        }
      >
        <View
          style={
            {
              "display": "flex",
              "flexDirection": "row",
              "fontWeight": "600",
              "gap": 20,
              "justifyContent": "space-between",
              "marginBottom": 20,
            }
          }
        >
          <Text
            style={
              {
                "fontFamily": "Inter",
                "fontSize": 15,
                "fontWeight": 600,
                "lineHeight": 20,
                "width": "50%",
              }
            }
          >
            Basic Salary
          </Text>
          <Text
            style={
              {
                "fontFamily": "Inter",
                "fontSize": 15,
                "fontWeight": 600,
              }
            }
          >
            GHS0.00
          </Text>
        </View>
        <View
          style={
            {
              "display": "flex",
              "flexDirection": "row",
              "fontWeight": "600",
              "gap": 20,
              "justifyContent": "space-between",
              "marginBottom": 20,
            }
          }
        >
          <Text
            style={
              {
                "fontFamily": "Inter",
                "fontSize": 15,
                "fontWeight": 600,
                "lineHeight": 20,
                "width": "50%",
              }
            }
          >
            Employee Pension Contribution Amount
          </Text>
          <Text
            style={
              {
                "fontFamily": "Inter",
                "fontSize": 15,
                "fontWeight": 600,
              }
            }
          >
            GHS0.00
          </Text>
        </View>
        <View
          style={
            {
              "display": "flex",
              "flexDirection": "row",
              "fontWeight": "600",
              "gap": 20,
              "justifyContent": "space-between",
              "marginBottom": 20,
            }
          }
        >
          <Text
            style={
              {
                "fontFamily": "Inter",
                "fontSize": 15,
                "fontWeight": 600,
                "lineHeight": 20,
                "width": "50%",
              }
            }
          >
            Employer Pension Amount
          </Text>
          <Text
            style={
              {
                "fontFamily": "Inter",
                "fontSize": 15,
                "fontWeight": 600,
              }
            }
          >
            GHS0.00
          </Text>
        </View>
        <View
          style={
            {
              "display": "flex",
              "flexDirection": "row",
              "fontWeight": "600",
              "gap": 20,
              "justifyContent": "space-between",
              "marginBottom": 20,
            }
          }
        >
          <Text
            style={
              {
                "fontFamily": "Inter",
                "fontSize": 15,
                "fontWeight": 600,
                "lineHeight": 20,
                "width": "50%",
              }
            }
          >
            Total PAYE Tax
          </Text>
          <Text
            style={
              {
                "fontFamily": "Inter",
                "fontSize": 15,
                "fontWeight": 600,
              }
            }
          >
            GHS0.00
          </Text>
        </View>
      </View>
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "alignItems": "center",
            "backgroundColor": "#0077B6",
            "borderRadius": 5,
            "borderWidth": 0,
            "display": "flex",
            "height": 38,
            "justifyContent": "center",
            "marginLeft": "auto",
            "marginRight": "auto",
            "opacity": 1,
            "width": 140,
          }
        }
      >
        <Text
          style={
            {
              "color": "white",
              "fontFamily": "Inter",
              "fontWeight": "600",
            }
          }
        >
          Calculate again
        </Text>
      </View>
    </View>
  </RCTScrollView>
</RCTSafeAreaView>
`;
